### ISSUE TITLE
New proxy preset for 3 common proxy headers

### ISSUE THREAD

**Issue by mholt** [mholt/caddy#851]
The [proxy directive](https://caddyserver.com/docs/proxy) has the notion of *presets* which are keywords that tune proxy's configuration a certain, pre-established way that is very common. Right now the only preset is websocket. But we should perhaps add one for these three lines:

```
proxy_header Host {host}
proxy_header X-Real-IP {remote}
proxy_header X-Forwarded-Proto {scheme}
```

These are so common that almost every web app requires them to function properly. I just used them to reverse proxy to a Discourse instance.

Should we make this a preset? And what should this preset be called?

And, this is very easy to do, should we agree to do it. Somebody who's new to the project or Go would like to take a stab at it?

**Comment by abiosoft** [comment]
Every single time I have used Caddy as a reverse proxy, I used those three lines too.

Why not just call it `reverse_proxy` ?

**Comment by mholt** [comment]
I always assumed the `proxy` directive *was* a `reverse_proxy` -- is it a particular kind of reverse proxy? Like a transparent reverse proxy or something?

But `reverse_proxy` isn't bad, if we can't think of anything else.

**Comment by abiosoft** [comment]
I would have preferred `reverse_proxy_headers` but that will be too long.

**Comment by dulldave** [comment]
Closing since it was fixed in #881.
